mdata/aggmetric.go

```diff
@@ -163,25 +163,38 @@ func (a *AggMetric) GetAggregated(consolidator consolidation.Consolidator, aggSp
 				badConsolidator.Inc()
 				return Result{}, err
 			case consolidation.Cnt:
-				agg = a.cntMetric
+				agg = aggregator.cntMetric
 			case consolidation.Lst:
-				agg = a.lstMetric
+				agg = aggregator.lstMetric
 			case consolidation.Min:
-				agg = a.minMetric
+				agg = aggregator.minMetric
 			case consolidation.Max:
-				agg = a.maxMetric
+				agg = aggregator.maxMetric
 			case consolidation.Sum:
-				agg = a.sumMetric
+				agg = aggregator.sumMetric
 			default:
 				err := fmt.Errorf("internal error: AggMetric.GetAggregated(): unknown consolidator %q", consolidator)
 				log.Errorf("AM: %s", err.Error())
 				badConsolidator.Inc()
 				return Result{}, err
 			}
 			if agg == nil {
-				return Result{}, fmt.Errorf("Consolidator %q not configured", consolidator)
+				return Result{}, fmt.Errorf("consolidator %q not configured", consolidator)
+			}
+			result, err := agg.Get(from, to)
+			if err != nil {
+				return Result{}, err
 			}
-			return agg.Get(from, to)
+			futurePoints := []schema.Point{}
+			if a.rob != nil {
+				futurePoints = a.rob.Get()
+				if len(futurePoints) > 0 && result.Oldest > futurePoints[0].Ts {
+					result.Oldest = futurePoints[0].Ts
```

Comment:
why? this is definitely not obvious and needs a comment.
But I disagree here.
In Foresee, we disregard all ROB samples that correspond to a boundary preceeding the aggregator boundary. and any sample read from agg.Get() would be one that has been flushed by the aggregator, ie samples that have a boundary lower than the boundary in the aggregator.

Thus, it follows that we wouldn't use a sample from futurePoints that is older than one present in result. Thus I don't think we should change the Oldest value here.

Reply:
What if all of the datapoints for this metric are in the `rob`?

Reply:
in that case, the adjustment makes sense.
I think we can make it work for both cases by only doing the adjustment when oldest is maxInt32 (kind of a hack), or better when `len(result.Iters) == 0`